sdk/core/Azure.Core/src/Shared/OperationPoller.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Implementation of LRO polling logic.
+    /// </summary>
+    internal class OperationPoller
```

Comment:
This type duplicates logic of both `OperationInternalBase` and `Operation`. What is the plan of consolidating all this?